hw/application_fpga/Makefile

```diff
@@ -381,41 +396,43 @@ application_fpga_testfw.bin: application_fpga.asc bram_fw.hex testfw.hex
 	@-$(RM) $<.tmp
 
 #-------------------------------------------------------------------
-# post-synthesis functional simulation.
+# Build testbench simulation for the design
 #-------------------------------------------------------------------
-synth_tb.vvp: $(P)/tb/tb_application_fpga.v synth.json
-	iverilog \
-		-o $@ \
-		-s tb_application_fpga synth.v $(P)/tb/tb_application_fpga.v \
+tb_application_fpga: $(SIM_VERILOG_SRCS) \
+			$(VERILOG_SRCS) \
+			$(PICORV32_SRCS) \
+			$(ICE40_SIM_CELLS) \
+			simfirmware.hex
+	python3 ./tools/app_bin_to_spram_hex.py \
+			./tb/app.bin \
+			./tb/output_spram0.hex \
+			./tb/output_spram1.hex \
+			./tb/output_spram2.hex \
+			./tb/output_spram3.hex \
+			|| { echo -e "\n -- Put your app.bin to simulate in the \"tb\" directory\n"; false; }
+	verilator \
+		-j $(shell nproc --ignore=1) \
+		--binary \
+		--cc \
+		--exe \
+		--Mdir tb_verilated \
+		--trace-fst \
+		--trace-structs \
+		--top-module tb_application_fpga_sim \
+		--timescale 1ns/1ns \
+		--timing \
+		-Wno-WIDTHEXPAND \
+		-Wno-UNOPTFLAT \
 		-DNO_ICE40_DEFAULT_ASSIGNMENTS \
-		$(ICE40_SIM_CELLS)
-	chmod -x $@
-
-synth_sim: synth_tb.vvp
```

Comment:
We have a few generated files mentioned in the .gitignore which I assume is from these once up on a time, but I'm not sure.
Such as:
```
/testbench.vvp
/testbench_wb.vvp
/testbench_ez.vvp
/testbench_sp.vvp
/testbench_rvf.vvp
/testbench_synth.vvp
/testbench.gtkw
/testbench.vcd
/testbench.trace
/testbench_verilator*
```